### PR TITLE
dash/editmode-module

### PR DESCRIPTION
**Pull request by sebastianbochan**
Extracted EditMode to separate module.

-----
- [x] - PoC and minimal demo
- [x] - error handling
- [x] - update demos
- [x] - update karma tests
- [x] - update cypress demos
- [x] - update docs

### CODE REVIEW

File: ts/masters-dashboards/modules/editmode.src.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
+/**
+ * @license Highcharts Dashboards Math @product.version@ (@product.date@)
```
Comment (karolkolodziej) [inline ts/masters-dashboards/modules/editmode.src.ts:2]: This needs to be changed.

File: ts/Dashboards/Board.ts [hunk 2]
```diff
@@ -179,16 +179,15 @@ class Board implements Serializable<Board, Board.JSON> {
...
         if (this.guiEnabled) {
             if (
-                EditMode && !(
+                Dashboards.EditMode && !(
                     this.options.editMode &&
                     !this.options.editMode.enabled
                 )
             ) {
-                this.editMode = new EditMode(this, this.options.editMode);
+                this.editMode = new Dashboards.EditMode(this, this.options.editMode);
```
Comment (karolkolodziej) [inline ts/Dashboards/Board.ts:187]: throw error when not defined

File: samples/dashboards/demo/minimal/demo.html [hunk 3]
```diff
@@ -1,6 +1,8 @@
 <script src="https://code.highcharts.com/highcharts.js"></script>
 <script src="https://code.highcharts.com/datagrid/datagrid.js"></script>
 <script src="https://code.highcharts.com/dashboards/dashboards.js"></script>
+<script src="https://code.highcharts.com/dashboards/modules/editmode.js"></script>
+<script src="https://code.highcharts.com/dashboards/modules/editmode.js"></script>
```
Comment (karolkolodziej) [inline samples/dashboards/demo/minimal/demo.html:5]: loaded twice

File: docs/dashboards/frequently-asked-questions.md [hunk 4]
```diff
@@ -15,6 +15,12 @@ After that, you need to pass the connector to the component config, and that’s
 How to connect component to a cell?
 ----------------------------------
 1. Enabled layout creator (GUI)
+To be able to use Dashboards with layout system and editmode you first have to load the editmode module.
```
Comment (karolkolodziej) [inline docs/dashboards/frequently-asked-questions.md:18]: Order matter- please mention this in the documentation.

File: ts/Dashboards/Board.ts [hunk 5]
```diff
@@ -178,17 +178,21 @@ class Board implements Serializable<Board, Board.JSON> {
...
         if (this.guiEnabled) {
-            if (
-                EditMode && !(
-                    this.options.editMode &&
-                    !this.options.editMode.enabled
-                )
-            ) {
-                this.editMode = new EditMode(this, this.options.editMode);
+            if (!Dashboards.EditMode) {
+                throw new Error('Missing editmode.js module');
```
Comment (karolkolodziej) [inline ts/Dashboards/Board.ts:182]: If you are using layouts, it is unclear if we should call it `editmode`... The `layout` sounds more intuitive.

**Review by karolkolodziej — CHANGES_REQUESTED** [review]
- [ ] Please go through demos- some of them are missing the `editmode` module- most likely because they are using the `layout`
- [x] We have a separate[ documentation about the edit mode](https://www.highcharts.com/docs/dashboards/edit-mode)- please modify it.
- [ ] `tools/gulptasks/dashboards/scripts-dts/modules/edit-mode.src.d.ts` is missing
- [ ] A lot of cypress tests are failing

File: docs/dashboards/frequently-asked-questions.md [hunk 6]
```diff
@@ -15,6 +15,14 @@ After that, you need to pass the connector to the component config, and that’s
...
```
Comment (karolkolodziej) [inline docs/dashboards/frequently-asked-questions.md:18]:
```suggestion
To be able to use Dashboards with layout system and edit mode you first have to load the `layout` module.
```

Comment (karolkolodziej) [reply docs/dashboards/frequently-asked-questions.md:18]: Similar change everywhere

**Review by karolkolodziej — CHANGES_REQUESTED** [review]

**Review by Vazonik — APPROVED** [review]

File: samples/dashboards/demo/live-weather/demo.html [hunk 7]
```diff
@@ -6,6 +6,7 @@
 <script src="https://code.highcharts.com/modules/solid-gauge.js"></script>
 <script src="https://code.highcharts.com/modules/accessibility.js"></script>
 <script src="https://code.highcharts.com/dashboards/dashboards.js"></script>
+<script src="https://code.highcharts.com/dashboards/modules/layout.js"></script>
```
Comment (jomar-honsi) [inline samples/dashboards/demo/live-weather/demo.html:9]: This demo doesn't use edit mode, layout.js should not be necessary.

**Review by jomar-honsi — APPROVED** [review]

File: samples/dashboards/demo/live-weather/demo.html (hunk 7 above)
Comment (sebastianbochan) [reply samples/dashboards/demo/live-weather/demo.html:9]: You setup the gui, which is a part of `layout.js`

Screenshot 2024-02-12 at 15 26 32

**Pull request merged**